web/struct_tree_layer_builder.js

```diff
@@ -156,6 +160,43 @@ class StructTreeLayerBuilder {
     }
   }
 
+  #addImageInTextLayer(node, element) {
+    const { alt, bbox, children } = node;
+    const child = children?.[0];
+    if (!this.#viewBox || !alt || !bbox || child?.type !== "content") {
+      return false;
+    }
+
+    const { id } = child;
+    if (!id) {
+      return false;
+    }
+    const span = document.getElementById(id);
+    if (!span) {
+      return false;
+    }
+
+    element.setAttribute("aria-owns", id);
+
+    const img = document.createElement("span");
+    span.append(img);
+    img.setAttribute("role", "img");
+    const label = removeNullCharacters(alt);
+    img.setAttribute("aria-label", label);
+
+    const { pageHeight, pageX, pageY } = this.#viewBox;
+    const rect = Util.normalizeRect(bbox);
+
+    const calc = "calc(var(--scale-factor)*";
+    const { style } = img;
+    style.width = `${calc}${rect[2] - rect[0]}px)`;
+    style.height = `${calc}${rect[3] - rect[1]}px)`;
+    style.left = `${calc}${rect[0] - pageX}px)`;
+    style.top = `${calc}${pageHeight - rect[3] + pageY}px)`;
```

Comment:
Does this handle the document being rotated in the viewer, since in that case we may re-use the same exact structTree?